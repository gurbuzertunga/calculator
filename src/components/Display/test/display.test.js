import { render, mount } from 'enzyme';
import { expect as chaiExpect } from 'chai';
import toJson from 'enzyme-to-json';
import Display from '../display';

describe('It works', () => {
  it('renders without crashing', () => {
    render(<Display />);
    const tree = shallow(<Display />);
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('renders the props correctly', () => {
    const display = mount(<Display op="op" next="next" total="total" />);
    chaiExpect(display.props().op).to.equal('op');
    chaiExpect(display.props().next).to.equal('next');
    chaiExpect(display.props().total).to.equal('total');
  });
  it('renders the content correctly', () => {
    const display = mount(<Display op="op" next="next" total="total" />);
    chaiExpect(display.text()).to.contain('total' && 'op' && 'next');
  });
});
