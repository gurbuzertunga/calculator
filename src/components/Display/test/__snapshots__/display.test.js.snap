// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It works renders without crashing 1`] = `
<div
  className="display w-100 d-flex flex-column justify-content-end align-items-center"
>
  <div
    className="total w-100 d-flex justify-content-end align-items-center"
  >
    0
  </div>
  <div
    className="op w-100 d-flex justify-content-end align-items-center"
  >
    0
  </div>
  <div
    className="next  w-100 d-flex justify-content-end align-items-center"
  >
    0
  </div>
</div>
`;
